docs/source/etcd-architecture.rst

```diff
@@ -0,0 +1,238 @@
+Calico network. This does not discuss the Calico etcd data model, which also
+acts as the primary API into the Calico network: for more on that, see
+:doc:`etcd-data-model`.
+
+.. _etcd: https://github.com/coreos/etcd
+
+Components
+----------
+
+Calico is made up of the following interdependent components:
+
+- :ref:`calico-felix-component`, the primary Calico agent that runs on each
+  machine that hosts endpoints.
+- :ref:`calico-orchestrator-plugin`, orchestrator-specific code that tightly
+  integrates Calico into that orchestrator.
+- :ref:`calico-etcd-component`, running unmodified and used by the other
+  components as a data store.
+- :ref:`calico-bgp-component`, a BGP client that distributes routing
+  information.
+- :ref:`calico-bgp-rr-component`, an optional BGP route reflector for higher
+  scale.
+
+The following sections break down each component in more detail.
+
+
+.. _calico-felix-component:
+
+Felix
+-----
+
+Felix is the most important component in the Calico network: without it, no
+network programming can be achieved. It is a daemon that runs on every machine
+that provides endpoints: in most cases that means on nodes that host containers
+or VMs.
+
+Depending on the specific orchestrator environment, Felix is responsible for
+some or all of the following tasks:
+
+Interface Creation and Management
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In some environments, particularly containerized ones, Felix is responsible
+for creating and managing network interfaces to its endpoints. This usually
+involves creating and managing 'veth pairs': a pair of virtual Ethernet
+interfaces that behave as though they're connected together with a single
+Ethernet cable.
+
+In all cases, regardless of what kind of environment it runs in, Felix also
+programs some other information about interfaces into the kernel. In
+particular, it may enable `Proxy ARP`_, `Proxy NDP`_, and IP forwarding for
+interfaces that it manages.
+
+.. _Proxy ARP: http://en.wikipedia.org/wiki/Proxy_ARP
+.. _Proxy NDP: http://en.wikipedia.org/wiki/Neighbor_Discovery_Protocol
+
+Route Programming
+~~~~~~~~~~~~~~~~~
+
+Felix is responsible for programming routes to the endpoints on its host into
+the Linux kernel FIB. This ensures that packets destined for those endpoints
+that arrive on at the host are forwarded accordingly.
+
+ACL Programming
+~~~~~~~~~~~~~~~
+
+Felix is also responsible for programming ACLs into the Linux kernel. These
+ACLs are used to ensure that only valid traffic can be sent between
+endpoints, and ensure that endpoints are not capable of circumventing
+Calico's security measures. For more on this, see :doc:`security-model`.
+
+State Reporting
+~~~~~~~~~~~~~~~
+
+Felix is responsible for providing data about the health of the network. In
+particular, it reports errors and problems with configuring its host. This data
+is written into etcd, to make it visible to other components and operators of
+the network.
```

Comment:
Very little of this document is true yet. I'd argue if we ship without this functionality we're being extremely reckless, so it doesn't feel aggressive to have this section in place.